File: src/main/java/com/distribution/modules/dis/controller/LoanOrderInfoController.java
package com.distribution.modules.dis.controller;

import com.distribution.common.utils.CommonUtils;
import com.distribution.common.utils.DateUtils;
import com.distribution.common.utils.Result;
import com.distribution.modules.dis.entity.LoanOrderInfoEntity;
import com.distribution.modules.dis.service.LoanOrderInfoService;
import com.github.pagehelper.PageHelper;
import com.github.pagehelper.PageInfo;
import lombok.extern.slf4j.Slf4j;
import org.apache.commons.collections.MapUtils;
import org.apache.shiro.authz.annotation.RequiresPermissions;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.Map;


/**
 * 贷款订单
 *
 * @author huchunliang
 * @email davichi2009@gmail.com
 * @date 2018-07-06
 */
@Slf4j
@RestController
@RequestMapping("loanorderinfo")
public class LoanOrderInfoController {
    @Autowired
    private LoanOrderInfoService loanOrderInfoService;

    /**
     * 列表
     */
    @RequestMapping("/list")
    @RequiresPermissions("loanorderinfo:list")
    public Result list(@RequestParam Map<String, Object> params) {
        //查询列表数据
        PageInfo<LoanOrderInfoEntity> pageInfo = PageHelper.startPage(MapUtils.getInteger(params, "page"),
                MapUtils.getInteger(params, "limit")).doSelectPageInfo(() -> loanOrderInfoService.queryList(params));
        return Result.ok().put("page", pageInfo);
    }


    /**
     * 信息
     */
    @RequestMapping("/info/{id}")
    @RequiresPermissions("loanorderinfo:info")
    public Result info(@PathVariable("id") String id) {
        LoanOrderInfoEntity loanOrderInfo = loanOrderInfoService.queryObject(id);

        return Result.ok().put("loanOrderInfo", loanOrderInfo);
    }
    @RequestMapping("/fenrunmoney")
    @RequiresPermissions("loanorderinfo:update")
    public Result huikuan(String applyforid, Double fenrunmoney) {
        LoanOrderInfoEntity loanOrderInfo = loanOrderInfoService.queryObject(applyforid);
        try {
            loanOrderInfo.setOrderStatus(1);
            loanOrderInfo.setLoanMoney(fenrunmoney);
            loanOrderInfo.setUpdateTime(DateUtils.formatDateTime(LocalDateTime.now()));
            loanOrderInfoService.update(loanOrderInfo);
        } catch (Exception e) {
            log.error("分润异常", e);
            return Result.error("分润异常");
        }
        return Result.ok();
    }
    /**
     * 保存
     */
    @RequestMapping("/save")
    @RequiresPermissions("loanorderinfo:save")
    public Result save(@RequestBody LoanOrderInfoEntity loanOrderInfo) {
        try {
            loanOrderInfo.setId(CommonUtils.getUUID());
            loanOrderInfo.setAddTime(DateUtils.formatDateTime(LocalDateTime.now()));
            loanOrderInfoService.save(loanOrderInfo);
        } catch (Exception e) {
            log.error("保存贷款订单信息异常", e);
            return Result.error("保存贷款订单信息异常");
        }

        return Result.ok();
    }

    /**
     * 修改
     */
    @RequestMapping("/update")
    @RequiresPermissions("loanorderinfo:update")
    public Result update(@RequestBody LoanOrderInfoEntity loanOrderInfo) {
        try {
            loanOrderInfo.setUpdateTime(DateUtils.formatDateTime(LocalDateTime.now()));
            loanOrderInfoService.update(loanOrderInfo);
        } catch (Exception e) {
            log.error("修改贷款订单信息异常", e);
            return Result.error("修改贷款订单信息异常");
        }
        return Result.ok();
    }

    /**
     * 删除
     */
    @RequestMapping("/delete")
    @RequiresPermissions("loanorderinfo:delete")
    public Result delete(@RequestBody String[] ids) {
        try {
            if (ids.length == 1) {
                loanOrderInfoService.delete(ids[0]);
            } else {
                loanOrderInfoService.deleteBatch(ids);
            }
        } catch (Exception e) {
            log.error("删除贷款订单信息异常", e);
            return Result.error("删除贷款订单信息异常");
        }
        return Result.ok();
    }

}

File: src/main/kotlin/com/distribution/modules/dis/controller/LoanOrderInfoController.kt
package com.distribution.modules.dis.controller

import com.distribution.common.utils.CommonUtils
import com.distribution.common.utils.DateUtils
import com.distribution.common.utils.Result
import com.distribution.modules.dis.entity.LoanOrderInfoEntity
import com.distribution.modules.dis.service.LoanOrderInfoService
import com.github.pagehelper.PageHelper
import org.apache.commons.collections.MapUtils
import org.apache.shiro.authz.annotation.RequiresPermissions
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*
import java.time.LocalDateTime


/**
 * 贷款订单
 *
 * @author huchunliang
 * @email davichi2009@gmail.com
 * @date 2018-07-06
 */

@RestController
@RequestMapping("loanorderinfo")
class LoanOrderInfoController {
    @Autowired
    private lateinit var loanOrderInfoService: LoanOrderInfoService
    private val log = LoggerFactory.getLogger(LoanOrderInfoController::class.java)

    /**
     * 列表
     */
    @RequestMapping("/list")
    @RequiresPermissions("loanorderinfo:list")
    fun list(@RequestParam params: Map<String, Any>): Result {
        //查询列表数据
        val pageInfo = PageHelper.startPage<Any>(MapUtils.getInteger(params, "page"),
                MapUtils.getInteger(params, "limit")).doSelectPageInfo<LoanOrderInfoEntity> { loanOrderInfoService.queryList(params) }
        return Result().ok().put("page", pageInfo)
    }


    /**
     * 信息
     */
    @RequestMapping("/info/{id}")
    @RequiresPermissions("loanorderinfo:info")
    fun info(@PathVariable("id") id: String): Result {
        val loanOrderInfo = loanOrderInfoService.queryObject(id)

        return Result().ok().put("loanOrderInfo", loanOrderInfo)
    }

    @RequestMapping("/fenrunmoney")
    @RequiresPermissions("loanorderinfo:update")
    fun huikuan(applyforid: String, fenrunmoney: Double): Result {
        val loanOrderInfo = loanOrderInfoService.queryObject(applyforid)
        try {
            loanOrderInfo.orderStatus = 1
            loanOrderInfo.loanMoney = fenrunmoney
            loanOrderInfo.updateTime = DateUtils.formatDateTime(LocalDateTime.now())
            loanOrderInfoService.update(loanOrderInfo)
        } catch (e: Exception) {
            log.error("分润异常", e)
            return Result().error(msg = "分润异常")
        }

        return Result().ok()
    }

    /**
     * 保存
     */
    @RequestMapping("/save")
    @RequiresPermissions("loanorderinfo:save")
    fun save(@RequestBody loanOrderInfo: LoanOrderInfoEntity): Result {
        try {
            loanOrderInfo.id = CommonUtils.uuid
            loanOrderInfo.addTime = DateUtils.formatDateTime(LocalDateTime.now())
            loanOrderInfoService.save(loanOrderInfo)
        } catch (e: Exception) {
            log.error("保存贷款订单信息异常", e)
            return Result().error(msg = "保存贷款订单信息异常")
        }

        return Result().ok()
    }

    /**
     * 修改
     */
    @RequestMapping("/update")
    @RequiresPermissions("loanorderinfo:update")
    fun update(@RequestBody loanOrderInfo: LoanOrderInfoEntity): Result {
        try {
            loanOrderInfo.updateTime = DateUtils.formatDateTime(LocalDateTime.now())
            loanOrderInfoService.update(loanOrderInfo)
        } catch (e: Exception) {
            log.error("修改贷款订单信息异常", e)
            return Result().error(msg = "修改贷款订单信息异常")
        }

        return Result().ok()
    }

    /**
     * 删除
     */
    @RequestMapping("/delete")
    @RequiresPermissions("loanorderinfo:delete")
    fun delete(@RequestBody ids: Array<String>): Result {
        try {
            when {
                ids.size == 1 -> {
                    val loanOrderInfoEntity = loanOrderInfoService.queryObject(ids[0])
                    loanOrderInfoEntity.isDelete = 0
                    loanOrderInfoService.update(loanOrderInfoEntity)
                }
                else -> ids.forEach {
                    val loanOrderInfoEntity = loanOrderInfoService.queryObject(it)
                    loanOrderInfoEntity.isDelete = 0
                    loanOrderInfoService.update(loanOrderInfoEntity)
                }
            }
        } catch (e: Exception) {
            log.error("删除贷款订单信息异常", e)
            return Result().error(msg = "删除贷款订单信息异常")
        }

        return Result().ok()
    }

}
